src/components/MovieListPage/movieListPage.jsx

```diff
@@ -23,17 +24,47 @@ const MovieListPage = (props) => {
     const [isLoading, setIsLoading] = useState(false);
     const [movies, setMovies] = useState([]); //movie list
 
+    const navigate = useNavigate();
+    const [searchParams, setSearchParams] = useSearchParams();
+
+    useEffect(() => {
+        setSearchParams({ sortCriterion, searchQuery, activeGenre });
+    }, [sortCriterion, searchQuery, activeGenre]);
+
+    useEffect(() => {
+        if(movieId) navigate(`/${movieId}`);
+    }, [movieId]);
+
+    useEffect(() => {
+      const params = new URLSearchParams(searchParams);
+      const newSortCriterion = params.get('sortCriterion') || 'release_date';
+      const newSearchQuery = params.get('searchQuery') || searchQuery;
+      const newActiveGenre = params.get('activeGenre') || 'All';
+      setSortCriterion(newSortCriterion);
+      setActiveGenre(newActiveGenre);
+      setSearchQuery(newSearchQuery);
+    }, [searchParams]);
+
     useEffect(() => {
         setIsLoading(true);
-        const query = buildQuery();
-        fetch(`http://localhost:4000/movies?${query}`)
-          .then(response => response.json())
-          .then(data => {
+        if(showDetail && movieId) {
+            fetch(`http://localhost:4000/movies/${movieId}`)
+            .then(response => response.json())
+            .then(data => {
+                setSelectedMovie(data[0]);
```

Comment:
this fetch is also redundant, setting selectedMovie and never using it

Reply:
Done.